packages/core/src/Stores/contract-store.js

```diff
@@ -462,6 +522,15 @@ function calculate_marker(contract_info, { accu_high_barrier, accu_low_barrier }
         if (exit_tick_time) {
             epoch_array.push(exit_tick_time);
         }
+        if (isResetContract(contract_type)) {
+            return {
+                contract_info: toJS(contract_info),
+                type: 'NonTickContract',
+                key: `${contract_id}-date_start`,
+                epoch_array: [reset_time, ...epoch_array],
+                price_array: [price_array[0], reset_barrier],
+            };
+        }
```

Comment:
```suggestion
```
no need for this, we can reassign epoch_array and price_array to what we need `if (isResetContract(contract_type)) {` above and it will work inside the same block below :)

Reply:
Can't understand your point ( This part is for non-tick contracts